File: 32.cpp
static bool compare(int a, int b)
{
    string strNum1=to_string(a);
    string strNum2=to_string(b);
    return (strNum1+strNum2)<(strNum2+strNum1);
}
string PrintMinNumber(vector<int> numbers) {
    string result;
    if(numbers.empty())
        return result;
    sort(numbers.begin(),numbers.end(),compare);
    for(unsigned int i=0;i<numbers.size();i++)
        result+=to_string(numbers[i]);
    return result;
}

File: 32.py
# -*- coding:utf-8 -*-
import itertools

class Solution:
    def PrintMinNumber(self, numbers):
        # write code here
        all = list(itertools.permutations(numbers))
        result = ''
        for i in range(len(all)):
            tmp = ''
            for n in all[i]:
                tmp += str(n)
            if i == 0 or tmp < result:
                result = tmp
        return result
S = Solution()
print(S.PrintMinNumber([3,32,321]))